picard/ui/widgets/editablelistview.py

```diff
@@ -134,6 +134,47 @@ def _move_rows_relative(self, rows, direction):
                 selection.setCurrentIndex(new_index, QtCore.QItemSelectionModel.Current)
 
 
+class UniqueEditableListView(EditableListView):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self._is_drag_drop = False
+
+    def setModel(self, model):
+        current_model = self.model()
+        if current_model:
+            current_model.dataChanged.disconnect(self.on_data_changed)
+        super().setModel(model)
+        model.dataChanged.connect(self.on_data_changed)
+
+    def dropEvent(self, event):
+        self._is_drag_drop = True
+        super().dropEvent(event)
+        self._is_drag_drop = False
+
+    def on_data_changed(self, top_left, bottom_right, roles):
+        # Do not drop duplicates during drag and drop as there is always
+        # a duplicate in the list for a short time in this case.
+        if self._is_drag_drop:
+            return
+        role = roles[0]
```

Comment:
Not sure if it is an issue or not, but https://doc.qt.io/qt-5/qabstractitemmodel.html#dataChanged says:
> An empty vector in the roles argument means that all roles should be considered modified. The order of elements in the roles argument does not have any relevance.

So it seems more correct to do:

```python
if QtCore.Qt.EditRole in roles:
   value = model.data(top_left, QtCore.Qt.EditRole)
```

Reply:
It's not really an issue because we control the model. But I change it anyway.